src/components/post-comment.jsx

```diff
@@ -72,6 +72,9 @@ export default class PostComment extends React.Component {
     if (shouldClearText) {
       this.setState({editText: ''});
     }
+    if (this.props.editText !== newProps.editText) {
```

Comment:
We store 'exact input state' in component and update redux version on blur, so we could restore 'new comment' version after cancelling comment. So, 1) props change in the same time redux-stored state change, which happens on things like 'someone clicked a baloon' or so; 2) not every props propagation to comment component is connected with comment text change, so we could be comparing old saved props with newly changed state, cancelling user's hard work on comment edit. Previous solution 'change state if props contain more long comment text' did quite bad here — not by function itself, but by it's complicated semantics.

So, for what I know now, checking props comment text change and updating component state after the check is correct thing to do.